addons/zeusFactory/functions/fnc_factoryLoop.sqf

```diff
@@ -0,0 +1,44 @@
+#include "script_component.hpp"
+
+params ["_args", "_pfID"];
+_args params ["_factory", "_side", "_newGroupParams", "_activeGroups"];
+if (isNull _factory) exitWith {
+    INFO_1("ending factory loop for logic [%1]",_factory);
+    [_pfID] call CBA_fnc_removePerFrameHandler;
+};
+
+// Filter groups
+_activeGroups = _activeGroups select {
+    (!isNull _x) && {({alive _x} count (units _x)) > 0}
+};
+_args set [3, _activeGroups];
+
+// Fill out current group in progress
+if (!(_newGroupParams isEqualTo [])) exitWith {
+    _newGroupParams params ["_placeLogic", "_group", "_soldierList"];
+    TRACE_3("_newGroupParams",_placeLogic,_group,_soldierList);
+
+    if (_soldierList isEqualTo []) then {
+        _args set [2, []]; // clear _newGroupParams
+        [_factory, _placeLogic, _group] call FUNC(dispatch);
+        _activeGroups pushBack _group;
+    } else {
+        private _unitType = _soldierList deleteAt 0;
+        private _unit = _group createUnit [_unitType, (getPos leader _group), [], 0, "FORM"];
```

Comment:
Is it possible for group locality to change during the spawning? If so createUnit will fail here.